tm2/pkg/bft/privval/signer/remote/client/client.go

```diff
@@ -0,0 +1,129 @@
+package client
+
+import (
+	"fmt"
+
+	r "github.com/gnolang/gno/tm2/pkg/bft/privval/signer/remote"
+	"github.com/gnolang/gno/tm2/pkg/bft/types"
+	"github.com/gnolang/gno/tm2/pkg/crypto"
+)
+
+// RemoteSignerClient type implements types.Signer.
+var _ types.Signer = (*RemoteSignerClient)(nil)
+
+// PubKey implements types.Signer.
+func (rsc *RemoteSignerClient) PubKey() (crypto.PubKey, error) {
```

Comment:
One of the very few instances where I agree with panicking instead of propagating an error

Reply:
We agreed not to fail in case of disconnection or server error but just to keep trying to reconnect to the server in a loop.
Also see this answer on the same topic: https://github.com/gnolang/gno/pull/3554#discussion_r2002111451

Reply:
The approach you suggested with prefetching the pubkey is good 👍